### PR TITLE
E - PLT-2094 Add custom branding functionality

### PR DESCRIPTION
**Pull request by jwilander**
Unit tests will be submitted against `plt-2057` branch when it gets rebased with these changes.

There is an enterprise PR to go along with this.

### CODE REVIEW

**Comment by coreyhulen** [comment]
+1

File: api/admin.go [hunk 1]
```diff
@@ -417,3 +421,118 @@ func getAnalytics(c *Context, w http.ResponseWriter, r *http.Request) {
...
+
+	if img, err := brandInterface.GetBrandImage(); err != nil {
+		w.Write(nil)
+	} else {
+		w.Header().Set("Content-Type", "image/png")
+		w.Write(img)
+	}
+}
+
+func saveBrandText(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (crspeller) [inline api/admin.go]: Seems like this should be more generic. There will be more text items in the future people will want to brand. For example, the helptext of the login field telling people to login with their "Company ID" which could be called something silly like their "Red ID" or whatever.

File: api/user.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func InitUser(r *mux.Router) {
 	sr.Handle("/verify_email", ApiAppHandler(verifyEmail)).Methods("POST")
 	sr.Handle("/resend_verification", ApiAppHandler(resendVerification)).Methods("POST")
 	sr.Handle("/mfa", ApiAppHandler(checkMfa)).Methods("POST")
-	sr.Handle("/generate_mfa_qr", ApiUserRequired(generateMfaQrCode)).Methods("GET")
+	sr.Handle("/generate_mfa_qr", ApiUserRequiredTrustRequester(generateMfaQrCode)).Methods("GET")
```
Comment (crspeller) [inline api/user.go]: Seems unrelated. Explain?

File: api/admin.go (hunk 1 above)
Comment (jwilander) [inline api/admin.go]: I imagine most of those would be configurable in the config file. The only reason I didn't put this in the config file was because it could potentially be pretty long and it's hard for a user to format text properly in the config file. I think this is the exception case where it's better in the DB.

0/5 though, it's pretty easy to generalize by making the request provide both key and value rather than just value

Comment (crspeller) [inline api/admin.go]: Now that you mention it. Why not just put this in the configuration file? Most users will just edit the text in the admin console so why the DB?

Comment (jwilander) [inline api/admin.go]: Hmmm, yeah I guess it doesn't really matter too much if it's in the config file since most users will edit it from the console. I'll move it over

**Comment by jwilander** [comment]
Ready for re-review @crspeller @coreyhulen 

I left the new sql_system_store.GetByName() function in on purpose since it's going to be useful in the future

**Comment by coreyhulen** [comment]
+1

**Comment by crspeller** [comment]
+1
